src/redux/actions/genes/updateGeneExpressionType.js

```diff
@@ -0,0 +1,33 @@
+import { GENES_EXPRESSION_TYPE_UPDATE } from '../../actionTypes/genes';
+import loadGeneExpression from './loadGeneExpression';
+
+const updateGeneExpressionType = (experimentId, expressionType) => async (dispatch, getState) => {
+  const {
+    views,
+  } = getState().genes.expression;
+
+  // Get components currently using the gene data, then update the components
+  const updates = Object.keys(views).map((componentUuid) => dispatch(
+    loadGeneExpression(
+      experimentId,
+      views[componentUuid].data,
+      componentUuid,
+      expressionType,
+      true,
+    ),
+  ));
```

Comment:
There is a better way to get all components that are currently using the gene expression data.

All expression data that is being visualised anywhere in the platform is stored under `genes/expression/data` in redux. Here is an example screenshot, where we have gene `FOS` displayed on the embedding and genes `S100A9`, `CD74` and `LYZ` displayed on the heatmap and the contents of the gene store:
<img width="1792" alt="Screenshot 2021-01-13 at 12 40 06" src="https://user-images.githubusercontent.com/65414651/104447725-791af100-559c-11eb-9931-bd3cba23ded1.png">

Therefore, in order to get all genes who need their expression updated, all you need to do is to get all gene names that are under `genes/expression/data`

Reply:
`loadGeneExpression` requires `genes` and `componentUuid` as a part of its parameters, that's why I used `genes/expression/views` to get `componentUuid` and the genes under each component and update them accordingly. I suppose we can use `genes/expression/data` directly and pass it into `loadGeneExpression`, and then use a preset `componentUuid` (e.g. 'InteractiveHeatmap'), but that would cause problems when a gene toggled for a panel (e.g. embedding) but not for other panel (e.g. heatmap) will show up in the wrong panel.

Else, I can try to update the genes directly without going through `loadGeneExpression`, e.g. refactor the data fetching part into a separate function, and have it under `loadGeneExpression` and `updateGeneExpressionType`.

Reply:
I agree with Agi, I just have one concern here regarding `expressionType` being in the root of `expressionType`. Let's discuss on a call.